cpp/cmake/thirdparty/get_gunrock.cmake

```diff
@@ -0,0 +1,64 @@
+#=============================================================================
+# Copyright (c) 2021, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#=============================================================================
+
+function(find_and_configure_gunrock VERSION)
+
+    if(NOT TARGET gunrock)
+        set(GUNROCK_GENCODE_SM60 OFF)
+        set(GUNROCK_GENCODE_SM61 OFF)
+        set(GUNROCK_GENCODE_SM70 OFF)
+        set(GUNROCK_GENCODE_SM72 OFF)
+        set(GUNROCK_GENCODE_SM75 OFF)
+        set(GUNROCK_GENCODE_SM80 OFF)
+
+        foreach(arch IN LISTS CMAKE_CUDA_ARCHITECTURES)
+            string(REPLACE "-real" "" ${arch} arch)
+            set(GUNROCK_GENCODE_SM${arch} "ON")
+        endforeach()
```

Comment:
Given that the CI build is only building for arch 80 instead of all supported, and this line is in the log:
```
-- Building for GPU_ARCHS = 60-real;70-real;75-real;80
```
I'm wondering if this code isn't working right (sorry I'm not more helpful here).  It's suspicious that arch 80 was enabled and it's the only one in the list above that doesn't have `-real;` attached to it.

For reference, here's the build log output for this build (which has failing HITS/Gunrock tests):
![image](https://user-images.githubusercontent.com/3039903/118154888-38fa3c80-b3dd-11eb-8fb5-3e3f81db0cef.png)

and here's the build log output for a successful CI build:
![image](https://user-images.githubusercontent.com/3039903/118154953-4a434900-b3dd-11eb-9127-7aa2377cdd87.png)

Reply:
Thanks for digging into this, and yes that is a bug. I will fix it asap

Reply:
I had inverted the `input_var` and `output_var` location for `string(REPLACE)`